docs/USAGE-GUIDE.md

```diff
@@ -495,6 +501,22 @@ To use OCSP stapling, both server and client must call `s2n_config_set_status_re
 
 The OCSP stapling information will be automatically validated if the underlying libcrypto supports OCSP validation. `s2n_config_set_check_stapled_ocsp_response()` can be called with "0" to turn this off. Call `s2n_connection_get_ocsp_response()` to retrieve the received OCSP stapling information for manual verification.
 
+#### CRL Validation
+
+> Note: the CRL validation feature in s2n-tls is currently considered unstable, meaning the CRL APIs are subject to change in a future release. To access the CRL APIs, include `api/unstable/crl.h`.
+
+Certificate Revocation Lists (CRLs) are lists of issued, unexpired certificates that have been revoked by the CA. CAs publish updated versions of these lists periodically. A validator wishing to verify a certificate obtains a CRL from the CA, validates the CRL, and checks to ensure the certificate is not contained in the list, and therefore has not been revoked by the CA.
+
+The s2n CRL lookup callback must be implemented and set via `s2n_config_set_crl_lookup_cb()` to enable CRL validation in s2n-tls. This callback will be triggered once for each certificate in the certificate chain. 
+
+The CRLs for all certificates received in the handshake must be obtained in advance of the CRL lookup callback invocation, externally from s2n-tls. Applications should load these CRLs into memory, either in advance or in the callback, by creating `s2n_crl`s via `s2n_crl_new()`, and adding the obtained CRL data via `s2n_crl_load_pem()`. The `s2n_crl` should be freed via `s2n_crl_free()` when no longer needed.
+
+The application must implement a way to look up the correct CRL for the given certificate. This lookup should use the hash of the received certificate's issuer, and the hash of the CRL's issuer, which will match. The certificate's issuer hash is retrieved via `s2n_crl_lookup_get_cert_issuer_hash()`, and the CRL's issuer hash is retrieved via `s2n_crl_get_issuer_hash()`. After the correct CRL is retrieved, call `s2n_crl_lookup_set()` to provide s2n-tls with this CRL, which will be used to validate the certificate with.
+
+If the CRL for a given certificate can't be found, call `s2n_crl_lookup_ignore()` to ignore this certificate. This will generally cause the certificate validation logic to fail with a `S2N_ERR_CRL_LOOKUP_FAILED` error. However, some received certificates are extraneous, and won't be included in the chain of trust. If an extraneous certificate is ignored, it will not cause the certificate validation logic to fail.
```

Comment:
```suggestion
Call `s2n_crl_lookup_ignore()` to ignore a received certificate if its CRL can't be found. This will cause the certificate validation logic to fail with a `S2N_ERR_CRL_LOOKUP_FAILED` error if the certificate is needed in the chain of trust. The certificate validation logic will not fail if the ignored certificate ends up not being included in the chain of trust.
```
Switching to more of an active voice.

Reply:
Thanks!!